engine/render/src/render/material.cpp

```diff
@@ -119,6 +119,12 @@ namespace dmRender
                 HConstant render_constant = dmRender::NewConstant(name_hash);
                 dmRender::SetConstantLocation(render_constant, location);
 
+                if (type == dmGraphics::TYPE_FLOAT_MAT4)
+                {
+                    num_values *= 4;
```

Comment:
"num_values" is the number of vector4's we need to allocate for in the constant buffer